cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/documentstore/DatabaseImpl.java

```diff
@@ -928,7 +890,7 @@ public Void call(SQLDatabase db) throws Exception {
                             if (newWinnerTx.getClass().equals(DocumentRevision.class)) {
                                 // user gave us a new DocumentRevision instance
                                 isNewOrUpdatedRevision = true;
-                            } else if (newWinnerTx.getClass().equals(InternalDocumentRevision.class)) {
+                            } else if (newWinnerTx instanceof InternalDocumentRevision) {
```

Comment:
Yes, this is ok unless we are concerned about sub-classes of `InternalDocumentRevision` that we should not cast or assume to be modified.